tools/lib/uobjnew.py

```diff
@@ -98,7 +98,16 @@
     u64 classptr = 0, size = 0;
     bpf_usdt_readarg(2, ctx, &classptr);
     bpf_usdt_readarg(4, ctx, &size);
-    bpf_probe_read_user(&key.name, sizeof(key.name), (void *)classptr);
+    u32 len32 = 0;
+    u64 len64 = 0;
+    size_t len = sizeof(key.name);
+    if (!bpf_usdt_readarg(3, ctx, &len32)) {
+      len = min(len, (size_t)len32);
+    } else {
+      bpf_usdt_readarg(3, ctx, &len64);
+      len = min(len, (size_t)len64);
+    }
```

Comment:
could you add some comments here esp. w.r.t. why attempt to read both u32 and u64 len?

Reply:
Updated, thanks